src/records.jl

```diff
@@ -83,17 +83,31 @@ Base.getindex(rec::Record, attr::Symbol) = getfield(rec, attr)
 Base.haskey(rec::T, attr::Symbol) where {T <: Record} = hasfield(T, attr)
 Base.keys(rec::T) where {T <: Record} = (fieldname(T, i) for i in 1:fieldcount(T))
 
-Base.start(rec::Record) = 0
-Base.done(rec::T, state) where {T <: Record} = state >= fieldcount(T)
+if isdefined(Base, :iterate)
+    function Base.iterate(rec::T, state=0) where T <: Record
+        state >= fieldcount(T) && return nothing
+        new_state = state + 1
+        return (fieldname(T, new_state) => getfield(rec, new_state), new_state)
+    end
 
-function Base.next(rec::T, state::Int) where T <: Record
-    new_state = state + 1
-    return (fieldname(T, new_state) => getfield(rec, new_state), new_state)
-end
+    function Base.iterate(rec::T, state=0) where T <: AttributeRecord
+        state >= fieldcount(T) && return nothing
+        new_state = state + 1
+        return (fieldname(T, new_state) => get(getfield(rec, new_state)), new_state)
+    end
```

Comment:
These two iterate methods have the same body. Combine with `T <: Union{Record, AttributeRecord}`

Reply:
Gross